crates/test-utils/src/bin/dump-katana.rs

```diff
@@ -23,11 +24,19 @@ async fn main() {
         .dump_state()
         .expect("Failed to call dump_state on Katana state");
 
-    let state = serde_json::to_string(&dump_state).expect("Failed to serialize state");
-
     // Dump the state
     std::fs::create_dir_all(".katana/").expect("Failed to create Kakata dump dir");
-    std::fs::write(".katana/dump.json", state).expect("Failed to write dump to .katana/dump.json");
+
+    let katana_dump_path = String::from(".katana/dump.json");
+    let katana_dump_file = File::options()
+        .create_new(true)
+        .read(true)
+        .write(true)
+        .append(false)
+        .open(&katana_dump_path)
+        .unwrap_or_else(|_| panic!("Failed to open file {}", &katana_dump_path));
```

Comment:
you can replace .unwrap_or_else(|_| panic!("Failed to open file {}", &katana_dump_path)) by `expect`

Reply:
@Eikix this is suggested by clippy.

The reason that I have found for this is because if we use `format!` in `expect`, then it is not lazily evaluated, that is format! will be called regardless of whether there is an error or not.

On the other hand `unwrap_or_else` is lazily evaluated.

Reply:
I see! fair